internal/serviceaccount/config.go

```diff
@@ -0,0 +1,229 @@
+package serviceaccount
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/yaml"
+)
+
+func createClusterRoleBinding(clientset *kubernetes.Clientset, namespace, serviceAccountName string, apply bool) (*rbacv1.ClusterRoleBinding, error) {
+	clusterRoleBinding := &rbacv1.ClusterRoleBinding{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "rbac.authorization.k8s.io/v1",
+			Kind:       "ClusterRoleBinding",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "method-read-only-binding",
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     "method-read-only-clusterrole",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      serviceAccountName,
+				Namespace: namespace,
+			},
+		},
+	}
+
+	ctx := context.Background()
+	if apply {
+		_, err := clientset.RbacV1().ClusterRoleBindings().Create(ctx, clusterRoleBinding, metav1.CreateOptions{})
+		if err != nil {
+			if errors.IsAlreadyExists(err) {
+				fmt.Println("ClusterRoleBinding 'method-read-only-binding' already exists, continuing...")
+				return clusterRoleBinding, nil
+			}
+			return nil, err
+		}
+	}
+	return clusterRoleBinding, nil
+}
+
+func createSecret(clientset *kubernetes.Clientset, namespace, serviceAccountName string, apply bool) (*corev1.Secret, error) {
+	secret := &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Secret",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "method-service-account-secret",
+			Namespace: namespace,
+			Annotations: map[string]string{
+				"kubernetes.io/service-account.name": serviceAccountName,
+			},
+		},
+		Type: corev1.SecretTypeServiceAccountToken,
+	}
+
+	ctx := context.Background()
+	if apply {
+		_, err := clientset.CoreV1().Secrets(namespace).Create(ctx, secret, metav1.CreateOptions{})
+		if err != nil {
+			if errors.IsAlreadyExists(err) {
+				fmt.Println("Secret 'method-service-account-secret' already exists, continuing...")
+				return secret, nil
+			}
+			return nil, err
+		}
+	}
+	return secret, nil
+}
+
+func createServiceAccount(clientset *kubernetes.Clientset, namespace, serviceAccountName string, apply bool) (*corev1.ServiceAccount, error) {
+	serviceAccount := &corev1.ServiceAccount{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "ServiceAccount",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceAccountName,
+			Namespace: namespace,
+		},
+	}
+
+	ctx := context.Background()
+	if apply {
```

Comment:
Having this in each function feels very duplicative. It seems like there isn't a great generic way to create resources, but that we could return a slice of interfaces and then do a switch based on type.